sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -78,6 +93,41 @@ public static SetAnalyzedStatement analyze(SetStatement node) {
         }
     }
 
+    public SetLicenseAnalyzedStatement analyze(SetLicenseStatement node, Analysis context) {
+        if (node.assignments().size() != SetLicenseAnalyzedStatement.LICENSE_TOKEN_NUM) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                "Invalid number of settings for SET LICENSE. Please provide %d settings.",
+                SetLicenseAnalyzedStatement.LICENSE_TOKEN_NUM));
+        }
+
+        final FieldProvider fieldProvider = new FieldProvider() {
+            @Override
+            public Symbol resolveField(QualifiedName qualifiedName, @Nullable List path, Operation operation) {
+                throw new UnsupportedOperationException("Cannot resolve field references");
+            }
+        };
+
+        ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
+            functions,
+            context.transactionContext(),
+            context.paramTypeHints(),
+            fieldProvider,
+            null);
+        Map<String, Symbol> statementSymbols = new HashMap<>(SetLicenseAnalyzedStatement.LICENSE_TOKEN_NUM);
+        ExpressionAnalysisContext exprCtx = new ExpressionAnalysisContext();
+
+        for (Assignment assignment : node.assignments()) {
+            String settingName = ExpressionToStringVisitor.convert(assignment.columnName(), Row.EMPTY);
+            if (!SetLicenseAnalyzedStatement.LICENSE_ALLOWED_TOKENS.contains(settingName)) {
+                throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                    "Invalid setting '%s' for SET LICENSE.", settingName));
+            }
+            Symbol valueSymbol = expressionAnalyzer.convert(assignment.expression(), exprCtx);
+            statementSymbols.put(settingName, valueSymbol);
+        }
+        return new SetLicenseAnalyzedStatement(statementSymbols);
```

Comment:
Unless I'm missing something, the current logic would allow providing the same license setting twice and pass all validations - probably resulting in a NPE down the way. (another reason to make the required arguments in the `SetLicenseAnalyzedStatement` explicit)